api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -50,6 +50,25 @@ type VolumeReplicationGroupSpec struct {
 	// Desired state of all volumes [primary or secondary] in this replication group;
 	// this value is propagated to children VolumeReplication CRs
 	ReplicationState volrep.ReplicationState `json:"replicationState"`
+
+	// S3 Endpoint to replicate PV metadata; set this field, along with a secret
+	// that contains the access-key-id and secret-access-key to enable VRG to
+	// replicate the PV metadata to the given s3 endpoint to a bucket with
+	// VRG's name.  Thus the VRG name needs to be unique among other VRGs using
+	// the same S3Endpoint and it should also follow AWS bucket naming rules:
+	// https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+	// If this field is not set, VRG may be used to simply control the replication
+	// state of all PVs in this group, under the expectation that the PV metadata
+	// is replicated by a different mechanism; this mode of operation may be
+	// referred to as backup-less mode.
+	S3Endpoint string `json:"s3Endpoint,omitempty"`
```

Comment:
S3Endpoint and S3SecretName should be required fields, right?

Reply:
Please see my comment above that provides a background for running VRG in a "backup-less" mode (also in the comments of volumereplicationgroup_types.go file).  Because of that, S3EndPoint and S3SecretName are optional as of now. If we decide that such a backup-less mode is no longer necessary, I agree with you that we should make these fields mandatory.